### ISSUE TITLE
Variable '$a' of type '[String]' where expected '[String]'. When String is input list of strings.

### ISSUE THREAD

**Issue by rustedsword** [graphql-perl/graphql-perl#9]
When using array of input types GraphQL always throws an exception about [Type] is not [Type] when passing variable data inside hashref to execute function.  If array of values embedded inside query itself then everything works fine (see query2 below).

Example script:
```
#!/usr/bin/env perl
use strict;
use warnings;
use GraphQL::Schema;
use GraphQL::Type::Object;
use GraphQL::Type::InputObject;
use GraphQL::Type::Scalar qw($String $Int);
use GraphQL::Execution qw(execute);
use Data::Dumper;

sub something_resolve {
	return { field1 => 'somestring' };
};

my $String_list = GraphQL::Type::List->new(of => $String);

my $something = {
	type => GraphQL::Type::Object->new(
		name => 'Something',
		fields => { field1 => { type => $String }}),
	args => { ttt => { type => $String_list } }
};

my $schema = GraphQL::Schema->new(
	query => GraphQL::Type::Object->new(
		name => 'Query',
		fields => { Something => $something }
	)
);

my $query = 'query ($a: [String]){
	Something(ttt: $a) { 
		field1
	}
}';

#my $query2 = 'query {
#	Something(ttt: ["first", "second"]) { 
#		field1
#	}
#}';

my $root_context = {Something => \&something_resolve };

my $a = execute($schema, $query, $root_context, undef, {a => ["first", "second"] } );
print Dumper($a);
```
Exception:
```
$VAR1 = {
          'errors' => [
                        {
                          'message' => 'Variable \'$a\' of type \'[String]\' where expected \'[String]\'.',
                          'path' => [
                                      'Something'
                                    ],
                          'locations' => [
                                           {
                                             'column' => 1,
                                             'line' => 5
                                           }
                                         ]
                        }
                      ],
          'data' => {
                      'Something' => undef
                    }
        };
```

**Comment by mohawk2** [comment]
This is a legitimate bug, which I will fix in a moment - thanks for the report!

This is the actual minimal repro case:

```perl
use Data::Dumper;
use GraphQL::Schema;
use GraphQL::Type::Scalar qw($String);
use GraphQL::Execution qw(execute);
my $schema = GraphQL::Schema->new(
  query => GraphQL::Type::Object->new(
    name => 'Query',
    fields => { 
      hello => { 
        type => $String,
        args => { arg => { type => GraphQL::Type::List->new(of => $String) } } 
      },
    } 
  ),
);
print Dumper execute(
  $schema, 'query q($a: [String]) {hello(arg: $a)}', { hello => "yo" },
  undef, { a => [ 'there' ] },
);
```

It works currently if instead of doing `GraphQL::Type::List->new(of => $String)` you give `$String->list`. The bug is in the type-comparison during query-execution.

**Comment by mohawk2** [comment]
Fixed and released as 0.27.

**Comment by rustedsword** [comment]
> instead of doing GraphQL::Type::List->new(of => $String) you give $String->list
Thanks, this is much better :)
